src/amber/controller/helpers/responders.cr

```diff
@@ -24,32 +24,51 @@ module Amber::Controller::Helpers
       end
 
       def js(js : String | ProcType)
```

Comment:
I'd suggest refactoring this with macros:

```cr
      {% for type in %w(html xml js json text) %}
        def {{type.id}}(value : String | ProcType)
          @available_responses[TYPE[:{{type.id}}]] = value
          self
        end

        def {{type.id}}(&block : -> _)
          {{type.id}}(block)
        end
      {% end %}

      def json(value : Hash(Symbol | String, String))
        json(value.to_json)
      end

      def json(**args : Object)
        json(args.to_h)
      end
```

Reply:
Less readable but definitely a good idea. I'll do that. 

While your here any idea why the specs won't pass in Travis. They work perfectly on all my local and work computers.

Reply:
I'm seeing `Error: formatting './src/amber/environment/settings.cr' produced changes`, could it be this?

Reply:
After this refactor `add_response` method becomes obsolete.

Reply:
Thank you. Interestingly Travis passed at the time. I guess format works differently in 0.27.1

I was still running format and specs with 0.27.0

Reply:
I've tried getting it to work with 0.27.2 now as well with no luck. Works locally but not on travis.